### DISCUSSION TITLE
Cache for subsequent gets on a resource

### DISCUSSION THREAD

**Discussion by mollstam** [concourse/concourse#8708] [Ideas]
Hello! 👋🏻 

I have a resource that fetches the latest version of our project from version control, for our build pipeline. It bothers me a bit that every new version requires a full download from version control. If I simply had a cache (on similar terms as a task cache would be great) I could download only the changes since the last version and possibly speed things up a bit.

Now its not a huge deal since version control and Concourse are running in the same data center but it kind of would _feel_ nice.

Is this possible today? Is it desirable to become possible? Or am I missing something making the whole idea void?

Thanks and make it a good day! ✨

**Comment by taylorsilva** [comment]
It's not possible and for this use case I don't think it would be desirable. Concourse doesn't want to make any assumptions about what can and can't be cached between different versions of your resource. So with a new `version` Concourse will create a fresh container for the `get` step to fetch that version.

Maybe you could have a "switch" on `get` steps to be cached, but that would required quite a few changes in various parts of Concourse in order to work. Immediate problem I see is that `get` steps aren't cached the same way `check` steps/jobs are. That'll probably be a thorny problem to untangle first.
